test/common/build/controllers.go

```diff
@@ -653,6 +649,10 @@ func RunBuildRunningPodDeleteTest(t testingT, clusterAdminClient *client.Client,
 	if newBuild.Status.Phase != buildapi.BuildPhaseError {
 		t.Fatalf("expected build status to be marked error, but was marked %s", newBuild.Status.Phase)
 	}
+
+	// allow time for the failed event to get created
+	time.Sleep(5 * time.Second)
```

Comment:
bump on this thread.  agree w/ @jim-minter that it looks like the 10s sleep can go.

for the 5s sleep, yeah a wait.Poll would be better, and i think we'd feel comfortable making it 30s or something to avoid flakes, too (Since we normally wouldn't wait the full 30s) then.